modules/test/jenkins-results-parser/src/main/java/com/liferay/jenkins/results/parser/BaseBuild.java

```diff
@@ -50,6 +50,63 @@
  */
 public abstract class BaseBuild implements Build {
 
+	@Override
+	public boolean applyReinvokeRules() {
+		if ((isCompleted() && !isFailing()) || !isCompleted() ||
+			isFromArchive() ||
+			(badBuildNumbers.size() >= REINVOCATIONS_SIZE_MAX)) {
+
+			return false;
+		}
+
+		for (ReinvokeRule reinvokeRule : ReinvokeRule.getReinvokeRules()) {
+			if (!reinvokeRule.matches(this)) {
+				continue;
+			}
+
+			reinvoke(reinvokeRule);
+
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override
+	public boolean applySlaveOfflineRules() {
+		if ((isCompleted() && !isFailing()) || !isCompleted() ||
+			isFromArchive()) {
+
+			return false;
+		}
+
+		JenkinsSlave jenkinsSlave = getJenkinsSlave();
+
+		if (jenkinsSlave == null) {
+			return false;
+		}
+
+		jenkinsSlave.update();
+
+		if (jenkinsSlave.isOffline()) {
+			return false;
+		}
+
+		for (SlaveOfflineRule slaveOfflineRule :
+				SlaveOfflineRule.getSlaveOfflineRules()) {
+
+			if (!slaveOfflineRule.matches(this)) {
+				continue;
+			}
+
+			takeSlaveOffline(slaveOfflineRule);
```

Comment:
@pyoo47 - I forgot to ask him, but it actually does apply the slave offline rules & reinvocation rules.  As seen in this portion.

I shouldn't have returned a boolean until it was actually going to be used.  But I think it'll be confusing if it's just a *isApply*.

I can change it back if this doesn't work.   Renaming isn't too difficult.

Reply:
I think Brian's definition of apply means that it has offlined the slave or reinvoked the build (depending on the type of rule), not evaluate. I think it's best if you change the naming back to what he set it to or talk it though with him before we send this in.